projects/zopfli/zopfli_compress_fuzzer.cc

```diff
@@ -0,0 +1,31 @@
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <string>
+
+#include "zopfli.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+
+  ZopfliOptions options;
+  ZopfliInitOptions(&options);
+
+  if (size == 0) {
```

Comment:
This library does not have a max length limit.

Reply:
The fuzz target can enforce it's own limit to avoid long inputs that are slow to process and may cause timeouts. Did you try running this fuzz target locally? What's an average speed you're observing?

Would be the best if you could do two runs:
1) without any corpus, just something like `./fuzzer -use_value_profile=1 -max_total_time=600 -print_final_stats=1`

2) with some corpus (e.g. grab source files of the project):

```
mkdir corpus
cp -r <source> corpus
./fuzzer -use_value_profile=1 -max_total_time=600 -print_final_stats=1 corpus
````

Just to see what execution speed is and whether long inputs make it very slow.

Reply:
With corpus:
Done 187 runs in 607 second(s)
stat::number_of_executed_units: 187
stat::average_exec_per_sec:     0
stat::new_units_added:          158
stat::slowest_unit_time_sec:    23
stat::peak_rss_mb:              742

WIthout corpus:
Done 65669 runs in 601 second(s)
stat::number_of_executed_units: 65669
stat::average_exec_per_sec:     109
stat::new_units_added:          1240
stat::slowest_unit_time_sec:    0
stat::peak_rss_mb:              471

Reply:
Thanks! That's what I meant -- with long inputs the target becomes unacceptably slow, 0 units per second on average. Let's cap it at `1024` maybe:

```
if (size ==  0 || size > 1024)
  return 0
```

Reply:
After capping it at 1024: 
It got faster on average, from 0 to 15 units per sec. 

With corpus:
Done 9523 runs in 601 second(s)
stat::number_of_executed_units: 9523
stat::average_exec_per_sec:     15
stat::new_units_added:          891
stat::slowest_unit_time_sec:    0
stat::peak_rss_mb:              616

Without corpus:
Done 66088 runs in 601 second(s)
stat::number_of_executed_units: 66088
stat::average_exec_per_sec:     109
stat::new_units_added:          1206
stat::slowest_unit_time_sec:    0
stat::peak_rss_mb:              475

Reply:
After capping 1024:

without corpus:
Done 66088 runs in 601 second(s)
stat::number_of_executed_units: 66088
stat::average_exec_per_sec:     109
stat::new_units_added:          1206
stat::slowest_unit_time_sec:    0
stat::peak_rss_mb:              475

with corpus:
Done 9523 runs in 601 second(s)
stat::number_of_executed_units: 9523
stat::average_exec_per_sec:     15
stat::new_units_added:          891
stat::slowest_unit_time_sec:    0
stat::peak_rss_mb:              616